graph-proxy/src/graphql/workflows.rs

```diff
@@ -347,4 +356,75 @@ impl WorkflowsQuery {
             .into_result()?;
         Ok(workflow.try_into()?)
     }
+
+    #[instrument(skip(self, ctx))]
+    async fn workflows(
+        &self,
+        ctx: &Context<'_>,
+        proposal_code: String,
+        proposal_number: u32,
+        visit: u32,
+        _continue: Option<String>,
+        limit: Option<u32>,
+    ) -> anyhow::Result<WorkflowList> {
+        let server_url = ctx.data_unchecked::<ArgoServerUrl>().deref();
+        let auth_token = ctx.data_unchecked::<Option<Authorization<Bearer>>>();
+        let namespace = format!("{}{}-{}", proposal_code, proposal_number, visit);
+        let client = reqwest::Client::new();
+        let request = |url: reqwest::Url| {
+            if let Some(auth_token) = auth_token {
+                client.get(url).bearer_auth(auth_token.token())
+            } else {
+                client.get(url)
+            }
+        };
+        let mut url = server_url.clone();
+        url.path_segments_mut()
+            .unwrap()
+            .extend(["api", "v1", "workflows", &namespace]);
+        let limit = limit.unwrap_or(10).min(10);
+        url.query_pairs_mut()
+            .append_pair("listOptions.limit", &limit.to_string());
+        if let Some(cursor) = _continue.clone() {
+            url.query_pairs_mut()
+                .append_pair("listOptions.continue", &cursor);
+        }
+        debug!("Retrieving workflows name from {url}");
+        let workflows_response = request(url)
+            .send()
+            .await?
+            .json::<APIResult<argo_workflows_openapi::IoArgoprojWorkflowV1alpha1WorkflowList>>()
+            .await?
+            .into_result()?;
+        let workflows =
+            future::try_join_all(workflows_response.items.into_iter().filter_map(|workflow| {
```

Comment:
Can we use a [`tokio::task::JoinSet`](https://docs.rs/tokio/latest/tokio/task/struct.JoinSet.html) here instead?